docs/contributing/index.rst

```diff
@@ -338,44 +559,84 @@ A good etiquette to take over is:
-the contributor become familiar with the contribution workflow, and for the core
-devs to become acquainted with the contributor; besides which, we frequently
-underestimate how easy an issue is to solve!
+The following are some guidelines on how new code should be written. Of
+course, there are special cases and there will be exceptions to these
+rules. However, following these rules when submitting new code makes
+the review easier so new code can be integrated in less time. Uniformly
+formatted code makes it easier to share code ownership.
 
-.. topic:: good first issue tag
+In addition to the PEP8 standards, geomstats follows the following
+guidelines:
 
-    A great way to start contributing to geomstats is to pick an item from
-    the list of `good first issues
-    <https://github.com/geomstats/geomstats/labels/good%20first%20issue>`_
-    in the issue tracker. Resolving these issues allow you to start contributing
-    to the project without much prior knowledge. If you have already contributed
-    to geomstats, you should look at Easy issues instead.
+1. Use underscores to separate words in non class names: ``n_samples``
+   rather than ``nsamples``.
 
-.. topic:: starter project tag
+2. Avoid single-character variable names. This prevents using automatic
+   tools to find-and-replace code, as searching for ``x`` in geomstats
+   will return the whole database. At least 3 characters are advised
+   for a variable name.
 
-    If you have already contributed to geomstats, another great way to contribute
-    to geomstats is to pick an item from the list of `starter project
-    <https://github.com/geomstats/geomstats/labels/starter%project>`_ in the issue
-    tracker. Your assistance in this area will be greatly appreciated by the
-    more experienced developers as it helps free up their time to concentrate on
-    other issues.
+3. Use understandable function and variable names. The naming should help
+   the maintainers reading faster through your code. Thus, ``my_array``,
+   ``aaa``, ``result``, ``res`` are generally bad variable names,
+   whereas ``rotation_vec`` or ``symmetric_mat`` read well.
 
-.. topic:: help wanted tag
+4. Avoid comment in the code, the documentation goes in the docstrings.
+   This allows the explanations to be included in the documentation
+   generated automatically on the website. Furthermore, forbidding comments
+   forces us to write clean code, and clean docstrings.
 
-    We often use the help wanted tag to mark issues regardless of difficulty. Additionally,
-    we use the help wanted tag to mark Pull Requests which have been abandoned
-    by their original contributor and are available for someone to pick up where the original
-    contributor left off. The list of issues with the help wanted tag can be found
-    `here <https://github.com/geomstats/geomstats/labels/help%20wanted>`__ .
+5. Follow geomstats' API. For example, points on manifolds are denoted
```

Comment:
capitalize Geomstats (I am going to stop commenting this, but maybe you can do a "command F" to fix in other places too)

Reply:
See above